code/modules/mob/inventory.dm

```diff
@@ -117,39 +107,45 @@ var/list/slot_equipment_priority = list( \
 //as they handle all relevant stuff like adding it to the player's screen and updating their overlays.
 
 //Returns the thing in our active hand
-/mob/proc/get_active_hand()
+/mob/proc/get_active_held_item()
 	RETURN_TYPE(/obj/item)
-	if(hand)	return l_hand
-	else		return r_hand
+	return null
 
-//Returns the thing in our inactive hand
-/mob/proc/get_inactive_hand()
-	if(hand)	return r_hand
-	else		return l_hand
+/mob/proc/get_active_held_item_slot()
+	return
 
-//Puts the item into your l_hand if possible and calls all necessary triggers/updates. returns 1 on success.
-/mob/proc/put_in_l_hand(var/obj/item/W)
-	if(lying || !istype(W))
-		return 0
-	return 1
+//Returns the thing in our inactive hand
+/mob/proc/get_inactive_held_items()
+	RETURN_TYPE(/list)
+	return null
 
-//Puts the item into your r_hand if possible and calls all necessary triggers/updates. returns 1 on success.
-/mob/proc/put_in_r_hand(var/obj/item/W)
-	if(lying || !istype(W))
-		return 0
-	return 1
+/mob/proc/get_held_items()
+	var/list/held_obj = get_inactive_held_items()
+	if(length(held_obj))
+		. = held_obj.Copy()
+	held_obj = get_active_held_item()
+	if(held_obj)
+		LAZYADD(., held_obj)
 
-//Puts the item into our active hand if possible. returns 1 on success.
+/mob/proc/get_empty_hand_slot()
+	return
+	
 /mob/proc/put_in_active_hand(var/obj/item/W)
-	return 0 // Moved to human procs because only they need to use hands.
+	. = equip_to_slot_if_possible(W, get_active_held_item_slot())
 
 //Puts the item into our inactive hand if possible. returns 1 on success.
 /mob/proc/put_in_inactive_hand(var/obj/item/W)
-	return 0 // As above.
+	. = equip_to_slot_if_possible(W, get_empty_hand_slot())
```

Comment:
Does this actually do what it says on the tin? Seems like it puts it in the first available empty hand slot, regardless of whether it's active.

Reply:
Added a `get_empty_hand_slots()` and a check for held item slot.